python/ray/dag/compiled_dag_node.py

```diff
@@ -1319,23 +1317,20 @@ def _get_or_compile(
             # The value of this dict is either the original channel or a newly
             # created CachedChannel (if the original channel is read more than once).
             channel_dict: Dict[ChannelInterface, ChannelInterface] = {}
+            # Handle non-input args
             for arg, consumers in arg_to_consumers.items():
```

Comment:
Hmm personally feel like it is cleaner we include input args and just do skip.

```python3
for arg, consumers in arg_to_consumers.items():

if isinstance(arg, InputNode) or isinstnace(arg, InputAttributeNode):
    continue

....
```

If you prefer this way, I think we should change the variable name to be bit more clear than `arg_to_consumers ` (but I couldn't find a good name for it, and that's why I prefer this way...).

Reply:
hmm, after the change of handling `num_input_consumers`, `args_to_consumers` is not used at all for input args. So I think there is no point adding it to `arg_to_consumers` and then skip if later.

There is already a comment update regarding `arg_to_consumers`:
# Dict from **non-dag-input arg** to the set of tasks that consume it. 
It should be very clear, together with the comment on L1320. 

Renaming a local variable to something unnecessarily long is not good for readability. But if there is a better name, we can certainly change.

Reply:
renamed to `non_input_arg_to_consumers`